### DISCUSSION TITLE
Pasting a line of text

### DISCUSSION THREAD

**Discussion by robwilkerson** [LazyVim/LazyVim#3554] [Q&A] [answered]
Long time casual user of Vim. I've long used it for editing a file here and there, interacting with config files, and other random editing tasks, but never as my daily coding driver.  I've recently become Neo-curious because these out-of-the box configs (and the Lua language) look like such a great way to really dig in full time without all of the fuss.

Anyway, grabbed NeoVim and cloned the LazyVim config  and I'm liking where I am...except...

Pasting a complete line of text doesn't work as expected. And by "as expected" I mean "like Vim". Or even like "NeoVim sans Lazy". I'm not sure what I could've done wrong, but this is a basic behavior change, so I'd be a little surprised if it was intentional. _Anyway_, if I cut/copy a complete line of text using either `dd` or `yy`, I'd expect to then be able to paste it as a completely new line of text using `p` to paste below the current line the cursor is on or `P` to paste above.

It doesn't happen that way for me. Both `p` and `P` paste to the right of the current cursor position on the same line. It behaves as though the new line is getting stripped, maybe?

As mentioned, it works as I'd expect in NeoVim sans Lazy, so it seems like it has to be something in the Lazy config, but as a new user, I'm not quite sure where to begin and would love some insight.

Is this intentional? Can I remap it?

Thanks for any help!

**Comment by dpetka2001** [comment]
[Screencast 2024-06-09 18:08:41.webm](https://github.com/LazyVim/LazyVim/assets/12776461/74240fb6-42b6-425c-9168-b7f1337d6c9e)

The default LazyVim starter template works correctly for me both with `dd/yy`.

**Reply by robwilkerson** [reply]
I must have something in conflict somewhere then, I guess. Will continue digging.

**Comment by folke** [comment]
You're probably the same person from that reddit post?
As I mentioned, LazyVim not Neovim change that behavior.

So it must be something in your config that's off.

You can debug this with:
* `:verbose nmap p`
* `:verbose nmap y`
* `:verbose nmap yy`

**Reply by robwilkerson** [reply]
I am, thanks for indulging me. This is really bugging me. I just pulled LazyVim fresh and am seeing the same behavior, but as soon as I delete the config it works as expected.  Here's what I see

`:verbose nmap p` --> `No mapping found`
`:verbose nmap y` --> 
<img width="807" alt="Screenshot 2024-06-09 at 11 24 05 AM" src="https://github.com/LazyVim/LazyVim/assets/7735/5654b0d7-0a71-474a-af7f-ef1562e75f36">
`:verbose nmap yy` --> 
<img width="709" alt="Screenshot 2024-06-09 at 11 27 00 AM" src="https://github.com/LazyVim/LazyVim/assets/7735/aa616015-0ec7-402f-a39f-2054179e62ba">

I do see a `WhichKey` conflict when I run `:LazyHealth`, but doesn't look directly related to cut/paste.

I hate being an outlier for such a common use case, but do appreciate the help.

**Reply by folke** [reply]
I can't reproduce this using a clean LazyVim install.
Try the docker installation and see for yourself.
Only thing I can think of is that your system clipboard is somehow acting up, but even then...

**Comment by robwilkerson** [comment]
I appreciate you at least looking into it. I clearly have something going on that isn't playing nice with one of the plugins, just can't imagine what it would be (yet). 

I'll have to check out my WezTerm config as well. I installed the Astro config over the weekend just to see and it behaved exactly the same way. Raw NeoVim behaves as expected, but as soon as I install a config, the inline pasting behavior resumes. If I find the key, I'll post here just to close the loop. Thank you again.

**Reply by folke** [reply, marked as answer]
Have you tried setting `vim.opt.clipboard = ""` in your `config/options.lua`?

**Reply by robwilkerson** [reply]
I had no idea that was a thing and I hadn't bumped into anything in my hours of searching that even hinted at such a setting, but I just installed it, opened a file, and 💥 ! It looks like that might be exactly what I needed! 

I'll have to dig in further to understand the change it's making and what it's unborking on my end, but I'm incredibly happy to have that basic behavior back for whatever reason it may've been gone. Thank you!

**Reply by folke** [reply]
So it seems your clipboard tool is the problem then.

Glad you at least found the culprit!

**Reply by robwilkerson** [reply]
Seems so, even if I don't yet understand why or which plugin it's interfering with. I'm on a Mac M2 laptop if that rings any bells for anyone, but yeah, looks like I'm back in business for all of the basics.

**Reply by muellert** [reply]
> Have you tried setting `vim.opt.clipboard = ""` in your `config/options.lua`?

I've long suffered this weird paste problem, but this solves it. Which docs should I have read, please?

**Comment by numbermess** [comment]
I wanted to chime in on this issues since I was running into the same problem. It turns out that my problem was caused by my desire to fix a different problem. I have installed an application called [Pure Paste](https://sindresorhus.com/pure-paste) that runs all the time in the background. It will remove formatted from text that has been copied to the clipboard. It works great in regular applications, but I think that it had an unintended side effect in Neovim.

I went through a lot of trial and error in my configuration files before suddenly remembering that I had this program. I tried turning it off and yanking/pasting a file and it looked like it worked as expected again. Running the program again re-introduced the issue of the yanked line getting pasted into the current cursor position.

After digging just a bit further I think I found where this problem was actually coming from. There is an option in Pure Paste that removes whitespace and line endings from copied text, and it looks like if this is enabled then your pasted content will be placed in the cursor's current position. When it's turned off the pasted content will go onto the next line.

With the Pure Paste "remove leading/trailing whitespace" option enabled, the pasted content appears at the cursor's position:
![screenshot-20250523-125837PM-wAWBwSYc@2x](https://github.com/user-attachments/assets/0e1c3a7d-b573-424e-af3d-58ddbcce79b8)

With the Pure Paste "remove leading/trailing whitespace" option disabled, the pasted content appears on the next line:
![screenshot-20250523-010040PM-iE6URQ5p@2x](https://github.com/user-attachments/assets/58bf3136-5a9f-4968-b419-8a4a2973d0f1)

Hopefully this will help somebody else coming across this.
